src/symmetric.jl

```diff
@@ -834,119 +838,73 @@ end
 ^(A::Symmetric{<:Complex}, p::Integer) = sympow(A, p)
 ^(A::SymTridiagonal{<:Real}, p::Integer) = sympow(A, p)
 ^(A::SymTridiagonal{<:Complex}, p::Integer) = sympow(A, p)
-function sympow(A::SymSymTri, p::Integer)
-    if p < 0
-        return Symmetric(Base.power_by_squaring(inv(A), -p))
-    else
-        return Symmetric(Base.power_by_squaring(A, p))
-    end
-end
-for hermtype in (:Symmetric, :SymTridiagonal)
-    @eval begin
-        function ^(A::$hermtype{<:Real}, p::Real)
-            isinteger(p) && return integerpow(A, p)
-            F = eigen(A)
-            if all(λ -> λ ≥ 0, F.values)
-                return Symmetric((F.vectors * Diagonal((F.values).^p)) * F.vectors')
-            else
-                return Symmetric((F.vectors * Diagonal(complex.(F.values).^p)) * F.vectors')
-            end
-        end
-        function ^(A::$hermtype{<:Complex}, p::Real)
-            isinteger(p) && return integerpow(A, p)
-            return Symmetric(schurpow(A, p))
-        end
-    end
-end
-function ^(A::Hermitian, p::Integer)
+^(A::Hermitian, p::Integer) = sympow(A, p)
+function sympow(A, p::Integer)
     if p < 0
-        retmat = Base.power_by_squaring(inv(A), -p)
+        return wrappertype(A)(Base.power_by_squaring(inv(A), -p))
```

Comment:
It would be good to have all such lines split over two, so that the changed line only represents the wrapper, and not the operation. A brief glance shows lines 844 and 846 as the only ones.